### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by jasonmattingly**
This PR goes hand in hand with https://github.com/Snapraise/Snap-Web/pull/1062 and they should be reviewed as complementary and released simultaneously.

**Accomplishes**

- [x] Snap abilities incorporated into the logic to restrict crud access to designated groups
- [x]  UI cleanups on query page, including rendering only those queries without custom variables in order to avoid confusion
- [x] blazer_groups added to flag queries and dashboard
- [x] validations added to require name and blazer_group for all dashboards and queries

### CODE REVIEW

File: app/controllers/blazer/queries_controller.rb [hunk 1]
```diff
@@ -3,17 +3,24 @@ class QueriesController < BaseController
...
       end
 
       if params[:filter] && params[:filter] != "dashboards"
         @dashboards = [] # TODO show my dashboards
       else
         @dashboards = Blazer::Dashboard.order(:name)
         @dashboards = @dashboards.includes(:creator) if Blazer.user_class
+
+        # Filters dashboards to only display those the user has access to
+        @dashboards =
```
Comment (vitchell) [inline app/controllers/blazer/queries_controller.rb:19]: this looks weird, not sure whats going on here or if its fully inteded

File: app/views/blazer/dashboards/show.html.erb [hunk 2]
```diff
@@ -25,14 +27,32 @@
   </p>
 <% end %>
 
-<%= render partial: "blazer/variables", locals: {action: dashboard_path(@dashboard)} %>
-
 <% @queries.each_with_index do |query, i| %>
+  <% blazer_group = BlazerGroup.find(query.blazer_group_id).try(:name).try(:to_sym) %>
```
Comment (vitchell) [inline app/views/blazer/dashboards/show.html.erb]: Find will raise an exception if the id is not there.  So the `try(:name)` here at least is unnecessary.  Don't overly use trys as that's a crutch and can lead to maintenance issues.

**Comment by vitchell** [comment]
We should probably make this private since we're adding changes to the gem that are not compatible with the original source. Its diverging into our own custom code.

**Comment by ddh** [comment]
@vitchell I agree that this should be private. it was in the works here: https://github.com/Snapraise/blazer-snap, until I ran into issues authorizing the github token in the gemfile without exposing sensitive oauth stuff in the Gemfile.lock. Maybe something you can help out with

File: app/controllers/blazer/queries_controller.rb (hunk 1 above)
Comment (jasonmattingly) [reply app/controllers/blazer/queries_controller.rb:19]: The main page normally displays all dashboards created. Even though users only have access to those dashboards we've given them access to, @DuyH also requested that users also only be able to see links to dashboards they can access, which is what this does. Some of the weirdness is due to trying to hide the names of our roles/abilities from this repo and keep that information in our own repo, which was also requested.

File: app/views/blazer/dashboards/show.html.erb (hunk 2 above)
Comment (jasonmattingly) [reply app/views/blazer/dashboards/show.html.erb]: Good point. I'm requiring blazer_group_ids and names anyway for queries and dashboards so it's definitely unnecessary, I'll remove the trys.

**Review by jonsrepo — APPROVED** [review]

**Review by ddh — APPROVED** [review]

**Pull request merged**